File: 04/pset4a.py
with open("input.txt", "r") as f:
    text = f.read().splitlines()

inputs = []
temp = ''

for i in text:
    if i == '':
        inputs.append(temp)
        temp = ''
        continue
    temp += " " + i
inputs.append(temp)

count = 0

for i in inputs:
    if 'byr:' in i and 'iyr:' in i and 'eyr:' in i and 'hgt:' in i and 'hcl:' in i and 'ecl:' in i and 'pid:' in i:
        count += 1

print(count)

File: 04/pset4a.js
const fs = require('fs');
const raw_input = fs.readFileSync('input.txt', {encoding: 'utf-8'}).split("\n").filter(x => x);
const lines = [];
const required = ['byr', 'iyr', 'eyr', 'hgt', 'hcl', 'ecl', 'pid'];

var input = '';
for (const line of raw_input) {
    if (line == '\r') {
        lines.push(input);
        input = '';
        continue;
    }
    input += line;
}
lines.push(input);

var count = 0;
for (const line of lines) {
    var i = 0;
    for (const item of required)
        if (line.split(item).length - 1) ++i;
    if (i >= 7) ++count;
}
console.log(count);